strider/trapi.py

```diff
@@ -390,3 +391,112 @@ def build_unique_kg_edge_ids(message: Message):
                 for eb in edge_binding_list:
                     if eb["id"] == edge_id:
                         eb["id"] = new_edge_id
+
+
+def is_valid_node_binding(message, nb, qgraph_node):
+    """
+    Check whether the given kgraph node
+    satisifies the given qgraph node
+    """
+    if qgraph_node.get("id", None) is not None:
+        if nb["id"] not in qgraph_node["id"]:
+            return False
+
+    kgraph_node = message["knowledge_graph"]["nodes"][nb["id"]]
+    if qgraph_node.get("category", None) is not None:
+        # Build list of allowable categories for kgraph nodes
+        qgraph_allowable_categories = []
+        for c in ensure_list(qgraph_node["category"]):
+            qgraph_allowable_categories.extend(
+                WBMT.get_descendants(c)
+            )
+
+        # Check that all categories on this
+        # kgraph node are allowed
+        if not all(
+            c in qgraph_allowable_categories
+            for c in kgraph_node["category"]
+        ):
+            return False
+    return True
+
+def is_valid_edge_binding(message, eb, qgraph_edge):
+    """
+    Check whether the given kgraph edge
+    satisifies the given qgraph edge
+    """
+    if qgraph_edge.get("id", None) is not None:
+        if eb["id"] not in qgraph_edge["id"]:
+            return False
+
+    kgraph_edge = message["knowledge_graph"]["edges"][eb["id"]]
+    if qgraph_edge.get("predicate", None) is not None:
+        # Build list of allowable predicates for kgraph edges
+        allowable_predicates = []
+        for p in ensure_list(qgraph_edge["predicate"]):
```

Comment:
If we add "predicate" to the `standardize_graph_lists` arguments, we wouldn't need to `ensure_list` here, right?

Reply:
My understanding is that knowledge graph edge predicates are not allowed to be lists. We could technically do that but it seems like we should keep our dictionaries TRAPI compliant.